app/models/audit/Auditable.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.audit
+
+import models.WithName
+import play.api.libs.json.{Json, Writes}
+
+trait Auditable { _: WithName =>
+  val auditDescription: String
+}
+
+object Auditable {
+  def writes[A]: Writes[A] = Writes {
```

Comment:
I had this originally, but it's not actually possible. Because, the top level `trait` which is the actual type doesn't extend the `Auditable` trait. It's only the individual case objects that can extend it because they're the only ones that can have the `WithName`. 

It's a tradeoff between enforcing `WithName` or making the generic more typesafe and I felt that the `WithName` won the argument. But I'm happy to be pursuaded otherwise.